### PR TITLE
[FLOC-1701] Automate testing redirects

### PR DESCRIPTION
**Pull request by adamtheturtle**
Fixes https://clusterhq.atlassian.net/browse/FLOC-1701

Some testing:

```
(flocker)~/Documents/ClusterHQ/flocker$ admin/test-redirects --doc-version=0.4.1dev1
(flocker)~/Documents/ClusterHQ/flocker$ admin/test-redirects --doc-version=0.4.0 --production
(flocker)~/Documents/ClusterHQ/flocker$ admin/test-redirects --doc-version=0.4.1 --production
'https://docs.clusterhq.com/' expected to redirect to 'https://docs.clusterhq.com/en/0.4.1/', instead redirects to 'https://docs.clusterhq.com/en/0.4.0/'.
'https://docs.clusterhq.com/en/latest/authors.html' expected to redirect to 'https://docs.clusterhq.com/en/0.4.1/authors.html', instead redirects to 'https://docs.clusterhq.com/en/0.4.0/authors.html'.
'https://docs.clusterhq.com/en/latest' expected to redirect to 'https://docs.clusterhq.com/en/0.4.1/', instead redirects to 'https://docs.clusterhq.com/en/0.4.0/'.
'https://docs.clusterhq.com/en/' expected to redirect to 'https://docs.clusterhq.com/en/0.4.1/', instead redirects to 'https://docs.clusterhq.com/en/0.4.0/'.
```

### CODE REVIEW

**Comment by wallrj** [comment]
I installed `Flocker` and `Flocker[release]` but got this when I run the tool for 0.4.1dev1

```
(1701)(redirect-FLOC-1701 ? =)[~/.../HybridLogic/flocker-2]$ admin/test-redirects --doc-version 0.4.1dev1
Traceback (most recent call last):
  File "admin/test-redirects", line 13, in <module>
    main(sys.argv[1:], top_level=TOPLEVEL, base_path=BASEPATH)
  File "/home/richard/projects/HybridLogic/flocker-2/admin/release.py", line 930, in test_redirects_main
    final_url = requests.get(original_url).url
  File "/home/richard/.virtualenvs/1701/lib/python2.7/site-packages/requests/api.py", line 60, in get
    return request('get', url, **kwargs)
  File "/home/richard/.virtualenvs/1701/lib/python2.7/site-packages/requests/api.py", line 49, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/richard/.virtualenvs/1701/lib/python2.7/site-packages/requests/sessions.py", line 457, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/richard/.virtualenvs/1701/lib/python2.7/site-packages/requests/sessions.py", line 569, in send
    r = adapter.send(request, **kwargs)
  File "/home/richard/.virtualenvs/1701/lib/python2.7/site-packages/requests/adapters.py", line 420, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: [Errno 1] _ssl.c:504: error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure

```

**Comment by wallrj** [comment]
And when trying to run this in the flocker-0.4.1dev branch using the script from another checkout with your branch, I get this error:


```
(1701)(release/flocker-0.4.1dev1 =)[~/.../HybridLogic/flocker-0.4.1dev1]$ ../flocker-2/admin/test-redirects --production
Traceback (most recent call last):
  File "../flocker-2/admin/test-redirects", line 13, in <module>
    main(sys.argv[1:], top_level=TOPLEVEL, base_path=BASEPATH)
  File "/home/richard/projects/HybridLogic/flocker-2/admin/release.py", line 901, in test_redirects_main
    options.parseOptions(args)
  File "/home/richard/.virtualenvs/1701/lib/python2.7/site-packages/twisted/python/usage.py", line 273, in parseOptions
    self.parseArgs(*args)
  File "/home/richard/projects/HybridLogic/flocker-2/admin/release.py", line 886, in parseArgs
    self['doc-version'] = get_doc_version(self['flocker-version'])
KeyError: 'flocker-version'

```

**Comment by wallrj** [comment]
Thanks @adamtheturtle 

The code looks pretty good and I like the simplified release process, but the tool didn't work for me.

Please resubmit when you've fixed it.

**Comment by adamtheturtle** [comment]
I think that it is fixed now.

**Comment by wallrj** [comment]
I still get the SSL handshake error.

`requests.exceptions.SSLError: [Errno 1] _ssl.c:504: error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure`

**Comment by adamtheturtle** [comment]
I run

```
mkvirtualenv 1701
pip install -e .[release]
admin/test-redirects --doc-version 0.4.1dev1
```

and get no error :disappointed:

**Comment by adamtheturtle** [comment]
OK I have recreated the issue on an inferior OS (:smile:) and have seemingly fixed it with a new dependency. Sorry for the back and forth, please try again.

File: admin/release.py [hunk 1]
```diff
@@ -934,6 +937,90 @@ def create_release_branch_main(args, base_path, top_level):
...
+
+            message = (
+                "'{original_url}' expected to redirect to '{expected_url}', "
+                "instead redirects to '{final_url}'.\n").format(
+                    original_url=original_url,
+                    expected_url=expected_url,
+                    final_url=final_url,
+            )
+
+            sys.stderr.write(message)
```
Comment (jongiddy) [inline admin/release.py]: Add a message if all the redirects pass, to confirm that the test passed successfully.

File: docs/gettinginvolved/infrastructure/release-process.rst [hunk 2]
```diff
@@ -190,27 +197,23 @@ Preparing For a Release
...
    .. prompt:: bash [vagrant@localhost]$
 
-      aws configure
+      ~/flocker-${VERSION}/admin/publish-docs --doc-version ${VERSION}
 
-#. Update the staging documentation:
+#. Check that the staging documentation is set up correctly:
 
-   This requires the BuildBot step to have finished.
+   The following command outputs error messages if the documentation does not redirect correctly.
```
Comment (jongiddy) [inline docs/gettinginvolved/infrastructure/release-process.rst]: Rephrase this once the positive message is also printed :question:

**Comment by jongiddy** [comment]
Confirm this works once the new dependency is added. Please address comments and merge.

**Comment by adamtheturtle** [comment]
I will merge this once the tests pass, thanks @jongiddy .

**Pull request merged**